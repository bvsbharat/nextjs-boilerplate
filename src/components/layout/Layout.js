import React from "react";
import PropTypes from "prop-types";
import Navbar from "../navbar";
import Footer from "../footer";

const Layout = ({ children }) => {
    return (
        <>
            <div className="absolute bg-gray-200 w-full h-full">
                <Navbar />
                <div className="container mx-auto  h-full  px-10 py-10">
                    {children}
                </div>
                <Footer />
            </div>
        </>
    );
};

Layout.propTypes = {
    children: PropTypes.element.isRequired,
};

export default Layout;
