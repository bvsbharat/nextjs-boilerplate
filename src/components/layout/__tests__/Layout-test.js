import React from "react";
import renderer from "react-test-renderer";
import Layout from "../Layout";

describe(Layout, () => {
    it("Should render correctly", () => {
        const component = renderer.create(
            <Layout>
                <h1>child</h1>
            </Layout>,
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
