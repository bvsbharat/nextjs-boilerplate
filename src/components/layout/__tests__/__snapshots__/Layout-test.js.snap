// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Should render correctly 1`] = `
<div
  className="absolute bg-gray-200 w-full h-full"
>
  <nav
    className="w-full mx-auto bg-white shadow"
  >
    <div
      className="max-w-7xl mx-auto px-2 sm:px-6 lg:px-8"
    >
      <div
        className="relative flex items-center justify-between h-16"
      >
        <div
          className="absolute inset-y-0 left-0 flex items-center sm:hidden"
        >
          <button
            className="inline-flex items-center justify-center p-2 rounded-md text-black-400 hover:text-white hover:bg-gray-700 focus:outline-none focus:ring-2 focus:ring-inset focus:ring-white"
            id="headlessui-disclosure-button-undefined"
            onClick={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            type="button"
          >
            <span
              className="sr-only"
            >
              Open main menu
            </span>
            <svg
              aria-hidden="true"
              className="block h-6 w-6"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4 6h16M4 12h16M4 18h16"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth={2}
              />
            </svg>
          </button>
        </div>
        <div
          className="flex-1 flex items-center justify-center sm:items-stretch sm:justify-start"
        >
          <div
            className="flex-shrink-0 flex items-center"
          >
            <h3
              className="text-base text-white font-bold tracking-normal leading-tight ml-3 block lg:hidden h-8 w-auto"
            >
              <span
                className="relative inline-block px-6 py-2"
              >
                <div
                  className="absolute inset-0 transform skew-x-12 bg-teal-accent-400"
                />
                <a
                  className="relative text-teal-900"
                  href="#"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  LOGO
                </a>
              </span>
            </h3>
            <h3
              className="text-base text-white font-bold tracking-normal leading-tight ml-3 hidden lg:block h-8 w-auto"
            >
              <span
                className="relative inline-block px-6 py-2"
              >
                <div
                  className="absolute inset-0 transform skew-x-12 bg-teal-accent-400"
                />
                <a
                  className="relative text-teal-900"
                  href="#"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  LOGO
                </a>
              </span>
            </h3>
          </div>
          <div
            className="hidden sm:block sm:ml-6"
          >
            <div
              className="flex space-x-4"
            >
              <a
                className="text-black-300 hover:text-teal-accent-400 px-3 py-2 rounded-md text-sm font-medium"
                href="#"
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                Some Link
              </a>
            </div>
          </div>
        </div>
        <div
          className="absolute inset-y-0 right-0 flex items-center pr-2 sm:static sm:inset-auto sm:ml-6 sm:pr-0"
        >
          <button
            className="bg-white-800 p-1 rounded-full text-black-400 hover:text-teal-accent-400 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-teal-accent-400 focus:ring-teal-accent-400"
          >
            <span
              className="sr-only"
            >
              View notifications
            </span>
            <svg
              aria-hidden="true"
              className="h-6 w-6"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15 17h5l-1.405-1.405A2.032 2.032 0 0118 14.158V11a6.002 6.002 0 00-4-5.659V5a2 2 0 10-4 0v.341C7.67 6.165 6 8.388 6 11v3.159c0 .538-.214 1.055-.595 1.436L4 17h5m6 0v1a3 3 0 11-6 0v-1m6 0H9"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth={2}
              />
            </svg>
          </button>
          <div
            className="ml-3 relative"
          >
            <div>
              <button
                aria-haspopup={true}
                className="bg-gray-800 flex text-sm rounded focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-teal-accent-400 focus:ring-white"
                id="headlessui-menu-button-1"
                onClick={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                type="button"
              >
                <span
                  className="sr-only"
                >
                  Open user menu
                </span>
                <img
                  alt=""
                  className="h-8 w-8 rounded"
                  src="https://images.unsplash.com/photo-1472099645785-5658abf4ff4e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=facearea&facepad=2&w=256&h=256&q=80"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </nav>
  <div
    className="container mx-auto  h-full  px-10 py-10"
  >
    <h1>
      child
    </h1>
  </div>
  <div
    className="bg-gray-900"
  >
    <div
      className="px-4 pt-16 mx-auto sm:max-w-xl md:max-w-full lg:max-w-screen-xl md:px-24 lg:px-8"
    >
      <div
        className="grid row-gap-10 mb-8 lg:grid-cols-6"
      >
        <div
          className="grid grid-cols-2 gap-5 row-gap-8 lg:col-span-4 md:grid-cols-4"
        >
          <div>
            <p
              className="font-medium tracking-wide text-gray-300"
            >
              Category
            </p>
            <ul
              className="mt-2 space-y-2"
            >
              <li>
                <a
                  className="text-gray-500 transition-colors duration-300 hover:text-deep-purple-accent-200"
                  href="/"
                >
                  News
                </a>
              </li>
              <li>
                <a
                  className="text-gray-500 transition-colors duration-300 hover:text-deep-purple-accent-200"
                  href="/"
                >
                  World
                </a>
              </li>
              <li>
                <a
                  className="text-gray-500 transition-colors duration-300 hover:text-deep-purple-accent-200"
                  href="/"
                >
                  Games
                </a>
              </li>
              <li>
                <a
                  className="text-gray-500 transition-colors duration-300 hover:text-deep-purple-accent-200"
                  href="/"
                >
                  References
                </a>
              </li>
            </ul>
          </div>
          <div>
            <p
              className="font-medium tracking-wide text-gray-300"
            >
              Apples
            </p>
            <ul
              className="mt-2 space-y-2"
            >
              <li>
                <a
                  className="text-gray-500 transition-colors duration-300 hover:text-deep-purple-accent-200"
                  href="/"
                >
                  Web
                </a>
              </li>
              <li>
                <a
                  className="text-gray-500 transition-colors duration-300 hover:text-deep-purple-accent-200"
                  href="/"
                >
                  eCommerce
                </a>
              </li>
              <li>
                <a
                  className="text-gray-500 transition-colors duration-300 hover:text-deep-purple-accent-200"
                  href="/"
                >
                  Business
                </a>
              </li>
              <li>
                <a
                  className="text-gray-500 transition-colors duration-300 hover:text-deep-purple-accent-200"
                  href="/"
                >
                  Entertainment
                </a>
              </li>
              <li>
                <a
                  className="text-gray-500 transition-colors duration-300 hover:text-deep-purple-accent-200"
                  href="/"
                >
                  Portfolio
                </a>
              </li>
            </ul>
          </div>
          <div>
            <p
              className="font-medium tracking-wide text-gray-300"
            >
              Cherry
            </p>
            <ul
              className="mt-2 space-y-2"
            >
              <li>
                <a
                  className="text-gray-500 transition-colors duration-300 hover:text-deep-purple-accent-200"
                  href="/"
                >
                  Media
                </a>
              </li>
              <li>
                <a
                  className="text-gray-500 transition-colors duration-300 hover:text-deep-purple-accent-200"
                  href="/"
                >
                  Brochure
                </a>
              </li>
              <li>
                <a
                  className="text-gray-500 transition-colors duration-300 hover:text-deep-purple-accent-200"
                  href="/"
                >
                  Nonprofit
                </a>
              </li>
              <li>
                <a
                  className="text-gray-500 transition-colors duration-300 hover:text-deep-purple-accent-200"
                  href="/"
                >
                  Educational
                </a>
              </li>
              <li>
                <a
                  className="text-gray-500 transition-colors duration-300 hover:text-deep-purple-accent-200"
                  href="/"
                >
                  Projects
                </a>
              </li>
            </ul>
          </div>
          <div>
            <p
              className="font-medium tracking-wide text-gray-300"
            >
              Business
            </p>
            <ul
              className="mt-2 space-y-2"
            >
              <li>
                <a
                  className="text-gray-500 transition-colors duration-300 hover:text-deep-purple-accent-200"
                  href="/"
                >
                  Infopreneur
                </a>
              </li>
              <li>
                <a
                  className="text-gray-500 transition-colors duration-300 hover:text-deep-purple-accent-200"
                  href="/"
                >
                  Personal
                </a>
              </li>
              <li>
                <a
                  className="text-gray-500 transition-colors duration-300 hover:text-deep-purple-accent-200"
                  href="/"
                >
                  Wiki
                </a>
              </li>
              <li>
                <a
                  className="text-gray-500 transition-colors duration-300 hover:text-deep-purple-accent-200"
                  href="/"
                >
                  Forum
                </a>
              </li>
            </ul>
          </div>
        </div>
        <div
          className="md:max-w-md lg:col-span-2"
        >
          <span
            className="text-base font-medium tracking-wide text-gray-300"
          >
            Subscribe for updates
          </span>
          <form
            className="flex flex-col mt-4 md:flex-row"
          >
            <input
              className="flex-grow w-full h-12 px-4 mb-3 transition duration-200 bg-white border border-gray-300 rounded shadow-sm appearance-none md:mr-2 md:mb-0 focus:border-deep-purple-accent-400 focus:outline-none focus:shadow-outline"
              placeholder="Email"
              required={true}
              type="text"
            />
            <button
              className="inline-flex items-center justify-center h-12 px-6 font-medium tracking-wide text-white transition duration-200 rounded shadow-md bg-deep-purple-accent-400 hover:bg-deep-purple-accent-700 focus:shadow-outline focus:outline-none"
              type="submit"
            >
              Subscribe
            </button>
          </form>
          <p
            className="mt-4 text-sm text-gray-500"
          >
            Bacon ipsum dolor amet short ribs pig sausage prosciuto chicken spare ribs salami.
          </p>
        </div>
      </div>
      <div
        className="flex flex-col justify-between pt-5 pb-10 border-t border-gray-800 sm:flex-row"
      >
        <p
          className="text-sm text-gray-500"
        >
          © Copyright 2020 Lorem Inc. All rights reserved.
        </p>
        <div
          className="flex items-center mt-4 space-x-4 sm:mt-0"
        >
          <a
            className="text-gray-500 transition-colors duration-300 hover:text-teal-accent-400"
            href="/"
          >
            <svg
              className="h-5"
              fill="currentColor"
              viewBox="0 0 24 24"
            >
              <path
                d="M24,4.6c-0.9,0.4-1.8,0.7-2.8,0.8c1-0.6,1.8-1.6,2.2-2.7c-1,0.6-2,1-3.1,1.2c-0.9-1-2.2-1.6-3.6-1.6 c-2.7,0-4.9,2.2-4.9,4.9c0,0.4,0,0.8,0.1,1.1C7.7,8.1,4.1,6.1,1.7,3.1C1.2,3.9,1,4.7,1,5.6c0,1.7,0.9,3.2,2.2,4.1 C2.4,9.7,1.6,9.5,1,9.1c0,0,0,0,0,0.1c0,2.4,1.7,4.4,3.9,4.8c-0.4,0.1-0.8,0.2-1.3,0.2c-0.3,0-0.6,0-0.9-0.1c0.6,2,2.4,3.4,4.6,3.4 c-1.7,1.3-3.8,2.1-6.1,2.1c-0.4,0-0.8,0-1.2-0.1c2.2,1.4,4.8,2.2,7.5,2.2c9.1,0,14-7.5,14-14c0-0.2,0-0.4,0-0.6 C22.5,6.4,23.3,5.5,24,4.6z"
              />
            </svg>
          </a>
          <a
            className="text-gray-500 transition-colors duration-300 hover:text-teal-accent-400"
            href="/"
          >
            <svg
              className="h-6"
              fill="currentColor"
              viewBox="0 0 30 30"
            >
              <circle
                cx="15"
                cy="15"
                r="4"
              />
              <path
                d="M19.999,3h-10C6.14,3,3,6.141,3,10.001v10C3,23.86,6.141,27,10.001,27h10C23.86,27,27,23.859,27,19.999v-10   C27,6.14,23.859,3,19.999,3z M15,21c-3.309,0-6-2.691-6-6s2.691-6,6-6s6,2.691,6,6S18.309,21,15,21z M22,9c-0.552,0-1-0.448-1-1   c0-0.552,0.448-1,1-1s1,0.448,1,1C23,8.552,22.552,9,22,9z"
              />
            </svg>
          </a>
          <a
            className="text-gray-500 transition-colors duration-300 hover:text-teal-accent-400"
            href="/"
          >
            <svg
              className="h-5"
              fill="currentColor"
              viewBox="0 0 24 24"
            >
              <path
                d="M22,0H2C0.895,0,0,0.895,0,2v20c0,1.105,0.895,2,2,2h11v-9h-3v-4h3V8.413c0-3.1,1.893-4.788,4.659-4.788 c1.325,0,2.463,0.099,2.795,0.143v3.24l-1.918,0.001c-1.504,0-1.795,0.715-1.795,1.763V11h4.44l-1,4h-3.44v9H22c1.105,0,2-0.895,2-2 V2C24,0.895,23.105,0,22,0z"
              />
            </svg>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
